File: 75.sort-colors/solution.go
func sortColors(nums []int)  {
    var count [3]int
    
    for _, n := range nums {
        count[n]++
    }
    nums = nums[:0]
    for i, n := range count {
        for range n {
            nums = append(nums, i)
        }
    }
}

File: 75.sort-colors/solution.kt
class Solution {
    fun sortColors(nums: IntArray) {
        val counts = nums.fold(arrayOf(0, 0, 0)) { acc, num -> acc.also { it[num]++ }}
        counts.withIndex().fold(0) { i, (color, count) -> 
            repeat(count) {
                nums[i + it] = color
            }
            i + count
        }
    }
}